actors/miner/src/state.rs

```diff
@@ -553,6 +553,46 @@ impl State {
         Ok(())
     }
 
+    #[allow(clippy::too_many_arguments)]
+    pub fn assign_sectors_to_deadline<BS: Blockstore>(
+        &mut self,
+        policy: &Policy,
+        store: &BS,
+        current_epoch: ChainEpoch,
+        mut sectors: Vec<SectorOnChainInfo>,
+        partition_size: u64,
+        sector_size: SectorSize,
+        deadline_idx: u64,
+    ) -> anyhow::Result<()> {
+        let mut deadlines = self.load_deadlines(store)?;
+        let mut deadline = deadlines.load_deadline(store, deadline_idx)?;
+
+        // Sort sectors by number to get better runs in partition bitfields.
+        sectors.sort_by_key(|info| info.sector_number);
+
+        if !deadline_is_mutable(
```

Comment:
just noting that this is the second time this is done during the NI onboarding flow, it's done prior to state modification and then in here where we're modifying state; I think that's OK because it's cheap?

Reply:
I added it to `assign_sectors_to_deadline` so it's safer function, and I've also added it to the beginning of `prove_commit_sectors_ni` so we exit early if deadline is not mutable. If we are ok with assumption that caller of `assign_sectors_to_deadline` will check this precondition, I'm fine with removing it from the function.

Reply:
Don't remove it here, I'd be more comfortable without the fail-early. But I think both is good.